app/src/main/java/org/simple/clinic/summary/PatientSummaryViewRenderer.kt

```diff
@@ -22,25 +22,25 @@ class PatientSummaryViewRenderer(
   }
 
   private fun setupUiForTeleconsult(model: PatientSummaryModel) {
-    if (model.isTeleconsultationEnabled) {
+    if (model.isTeleconsultationEnabled && model.isUserLoggedIn) {
```

Comment:
There is a potential race condition here since the only condition for calling this method is if we have fetched the current facility info. If the facility gets fetched before the current user and a model update is triggered, this might cause issues.

We should probably also add another check if the logged in status has been loaded or not.

Reply:
Related: https://github.com/simpledotorg/simple-android/pull/1321#discussion_r425760779

Reply:
Yeah, I am worried about this. One option I considered was loading the facility and user simultaneously and use that even to start the effect and update the model as well. But was sure if that a good way to do in Mobius. We can discuss this on Monday 👍